CMakeLists.txt

```diff
@@ -29,14 +29,14 @@ ENDIF()
 ADD_DEFINITIONS(-DHARDWARE_ID="${HARDWARE_ID}")
 message(STATUS "Setting HARDWARE_ID to ${HARDWARE_ID}")
 
-IF (NOT DEFINED DEVICE_STREAMS)
-    SET(DEVICE_STREAMS "release,premerge")
+IF (NOT DEFINED DEVICE_FACTORY)
+    message(FATAL_ERROR "Missing required value: DEVICE_FACTORY")
 ENDIF()
-ADD_DEFINITIONS(-DDEVICE_STREAMS="${DEVICE_STREAMS}")
-message(STATUS "Setting DEVICE_STREAMS to ${DEVICE_STREAMS}")
+ADD_DEFINITIONS(-DDEVICE_FACTORY="${DEVICE_FACTORY}")
+message(STATUS "Setting DEVICE_FACTORY to ${DEVICE_FACTORY}")
```

Comment:
We need to be careful here. I'm adding @ricardosalveti. The problem is that this breaks how our current customers do their overrides. eg
~~~
# meta-subscriber-overrides
$ cat recipes-sota/lmp-device-register/lmp-device-register_git.bbappend 
EXTRA_OECMAKE += "-DDEVICE_STREAMS=andy-corp -DDEVICE_API=https://api.foundries.io/ota/devices/ -DGIT_COMMIT=${SRCREV} -DHARDWARE_ID=${MACHINE}"
~~~

The question is: If we break compatibility here, can we make our bitbake recipe smart enough to transform a customer's `-DDEVICE_STREAMS=andy-corp` into `-DDEVICE_FACTORY=andy-corp`

Reply:
This will indeed require a factory update for every user, since it gets defined via bbappend. Maybe keep DEVICE_STREAMS temporarily as a deprecated settings? This should give us time to update/fix all factories.

Reply:
could we do some logic in our bitbake recipe to convert DEVICE_STREAMS into DEVICE_FACTORY automagically? Or does that get too crazy?

Reply:
The 'if' logic can be driven by some value fetched from the lmp-device-register repo, maybe the latest tag.

Reply:
taking into account the plan to move lmp-device-register to aktualizr-lite, what will require updating recipes anyway, it looks like supporting both DEVICE_STREAMS and DEVICE_FACTORY in the make looks like the most practical at the moment.

Reply:
> could we do some logic in our bitbake recipe to convert DEVICE_STREAMS into DEVICE_FACTORY automagically? Or does that get too crazy?

I see what you are saying here, we can probably try something in the lmp-device-register recipe in python, to replace the variable before calling do configure (something like`FOO ??= "${@d.getVar('FOO').replace('BAR', 'FOO')}"`.

Reply:
> taking into account the plan to move lmp-device-register to aktualizr-lite, what will require updating recipes anyway, it looks like supporting both DEVICE_STREAMS and DEVICE_FACTORY in the make looks like the most practical at the moment.

Indeed, let's support both for now and do the proper migration when we move this to ak-lite.

Reply:
> Indeed, let's support both for now and do the proper migration when we move this to ak-lite.

Done